ray-operator/controllers/ray/utils/dashboard_httpclient.go

```diff
@@ -222,6 +223,7 @@ type RayJobLogsResponse struct {
 }
 
 // Note that RayJobInfo and error can't be nil at the same time.
+// Please make sure if the Ray job with JobId can't be found. Return a BadRequest error.
```

Comment:
This is somewhat unusual, but the function NewNotFound requires schema.GroupResource as a parameter.

```go
func NewNotFound(qualifiedResource schema.GroupResource, name string) *StatusError
```